src/librustc_resolve/build_reduced_graph.rs

```diff
@@ -614,7 +614,11 @@ impl<'a, 'cl> Resolver<'a, 'cl> {
 
     /// Builds the reduced graph for a single item in an external crate.
     fn build_reduced_graph_for_external_crate_def(&mut self, parent: Module<'a>, child: Export) {
-        let Export { ident, def, vis, span, .. } = child;
+        let Export { ident, def, vis, span } = child;
+        // FIXME: We shouldn't create the gensym here, it should come from metadata,
+        // but metadata cannot encode gensyms currently, so we create it here.
+        // This is only a guess, two equivalent idents may incorrectly get different gensyms here.
```

Comment:
Is this even a hypothetical problem for `_`?

Reply:
I'm... not entirely sure.
I haven't found an example where it would caused problems so far (assuming gensym creation is delayed until name resolution).

Reply:
https://github.com/rust-lang/rust/pull/56303#issuecomment-442464695 has an example of gensyms being created incorrectly internally, but it's not observable from the outside.